odoo/modules/loading.py

```diff
@@ -387,6 +387,12 @@ def load_modules(db, force_demo=False, status=None, update_module=False):
         for package in graph:
             migrations.migrate_module(package, 'end')
 
+        # STEP 3.6: warn about missing NOT NULL constraints
+        for (table, column), exc in registry._notnull_errors.items():
+            msg = "Table %r: column %r: unable to set constraint NOT NULL"
+            _logger.warning(msg, table, column, exc_info=exc)
```

Comment:
I'm not sure this works correctly in Python 2 (which I think 11 was still supposed to be at least compatible with?), as there the traceback is not part of the exception object.

Reply:
I thought about that case, too.  My guess is that it will produce no traceback at all, which is not that problematic, since the message is clear enough IMHO.  Maybe we should test it, just to make sure it does not miserably crash.